src/Psr/Tracing/RelayOpenTelemetry.php

```diff
@@ -0,0 +1,130 @@
+<?php
+
+declare(strict_types=1);
+
+namespace CacheWerk\Relay\Psr\Tracing;
+
+use Relay\Relay;
+
+use OpenTelemetry\API\Trace\SpanKind;
+use OpenTelemetry\API\Trace\TracerInterface;
+
+class RelayOpenTelemetry
+{
+    /**
+     * The Relay instance.
+     *
+     * @var \Relay\Relay
+     */
+    protected Relay $relay;
+
+    /**
+     * The OpenTelemetry tracer instance.
+     *
+     * @var \OpenTelemetry\API\Trace\TracerInterface
+     */
+    protected TracerInterface $tracer;
+
+    /**
+     * Creates a new `NewRelicRelay` instance.
+     *
+     * @param  \Relay\Relay  $relay
+     * @param  \OpenTelemetry\API\Trace\TracerInterface  $tracer
+     * @return void
+     */
+    public function __construct(Relay $relay, TracerInterface $tracer)
```

Comment:
I would recomment to accept a `TracerProvider` instead of a `TracerInterface`, because it is up to an instrumentation how to name and version tracers.

You should create here a tracer passing to it the Relay version, e.g.:

```php
$this->tracer = $tracerProvider->getTracer('github.com/cachewerk/relay', $relayVersion);
```

In future, even the `$tracerProvider` might be optional because there likely will be a global/default tracer provider.